client/src/main/java/com/paypal/selion/internal/platform/grid/LocalHub.java

```diff
@@ -44,7 +45,11 @@ synchronized LocalHub getLocalServerComponent() {
             instance = new LocalHub();
 
             instance.setHost(new NetworkUtils().getIpOfLoopBackIp4());
-            instance.setPort(Integer.parseInt(Config.getConfigProperty(ConfigProperty.SELENIUM_PORT)));
+            
+            int configHubPort = Integer.parseInt(Config.getConfigProperty(ConfigProperty.SELENIUM_PORT));
+            // Choose another free port when config port is not available.
+            int hubPort = PortProber.pollPort(configHubPort) ? configHubPort : PortProber.findFreePort();
+            instance.setPort(hubPort);
```

Comment:
Should we update the value of the property SELENIUM_PORT if we are not using 4444?